specification/digitaltwins/resource-manager/Microsoft.DigitalTwins/stable/2022-10-31/digitaltwins.json

```diff
@@ -0,0 +1,2299 @@
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "DigitalTwins instance properties.",
+          "$ref": "#/definitions/DigitalTwinsProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/DigitalTwinsResource"
+        }
+      ]
+    },
+    "DigitalTwinsPatchDescription": {
+      "description": "The description of the DigitalTwins service.",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "description": "Instance patch properties",
+          "type": "object",
+          "x-nullable": true,
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "identity": {
+          "description": "The managed identity for the DigitalTwinsInstance.",
+          "x-nullable": true,
+          "$ref": "#/definitions/DigitalTwinsIdentity"
+        },
+        "properties": {
+          "description": "Properties for the DigitalTwinsInstance.",
+          "x-nullable": true,
+          "$ref": "#/definitions/DigitalTwinsPatchProperties"
+        }
+      }
+    },
+    "DigitalTwinsResource": {
+      "description": "The common properties of a DigitalTwinsInstance.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource name.",
+          "pattern": "^(?!-)[A-Za-z0-9-]{3,63}(?<!-)$"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location.",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "type": "object",
+          "x-nullable": true,
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The resource tags."
+        },
+        "identity": {
+          "description": "The managed identity for the DigitalTwinsInstance.",
+          "x-nullable": true,
+          "$ref": "#/definitions/DigitalTwinsIdentity"
+        },
+        "systemData": {
+          "description": "Metadata pertaining to creation and last modification of the DigitalTwinsInstance.",
+          "readOnly": true,
+          "$ref": "#/definitions/SystemData"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "required": [
+        "location"
+      ]
+    },
+    "DigitalTwinsIdentity": {
+      "description": "The managed identity for the DigitalTwinsInstance.",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "None",
```

Comment:
Any reason you are not using the common types : 

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json)

Reply:
All other issues could be avoided if you were using common types from the beginning

Reply:
If we start using these now, wouldn't that cause a breaking change? and not only here, but also in management SDKs?)

Reply:
Identity is a standard contract across all azure resources. Unless you are using any additional custom field it should not cause any breaking change

Reply:
We tried updating to the common type but unfortunately it it will cause breaking changes in our SDKs. We would prefer to stick with the existing types